### PR TITLE
Pool: adds a default param for max_slippage_bps when initializing xyk pool

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

**Comment by gangov** [comment]
adds a new input parameter to `pool` `initialize` function that will be stored as a variable inside config, called `default_slippage_bps`.

we use that for default, instead of `100` when the customer hasn't specified any slippage.

besides these changes, this PR lints and fixes tests inside factory

File: contracts/factory/src/contract.rs [hunk 1]
```diff
@@ -188,6 +188,10 @@ impl FactoryTrait for Factory {
         )
             .into_val(&env);
 
+        if let PoolType::Xyk = pool_type {
+            init_fn_args.push_back(100i64.into_val(&env));
```
Comment (ueco-jb) [inline contracts/factory/src/contract.rs:192]: But now you've hardcoded it for the factory. It should becoma a parameter during the pool creations.

Comment (gangov) [reply contracts/factory/src/contract.rs]: jeez, I guess I thought I was inside the test setup. Fixed now

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**